en/docs/administration/production-deployment-guideline.md

```diff
@@ -0,0 +1,72 @@
+
+
+## Change the hostnames and vhosts
+
+By default, APK uses wso2.com for its hostnames and vhosts for the gateway. You need to change these values to your own domain, which you plan to use for production. The following values.yaml values should be modified:
+
+- wso2.apk.listener.hostname
+- wso2.apk.dp.gateway.listener.hostname
+- wso2.apk.dp.configdeployer.vhosts
+
+For further clarification on the keys, please refer to the description and default values [here](https://github.com/wso2/apk/blob/main/helm-charts/README.md)
+
+## Change the gateway certificates
+
+The default APK deployment uses a self-signed certificate for the gateway. You need to update it with your domain-validated certificate through the `values.yaml` file.
+
+Prerequisites
+
+1. TLS certificate verified by a Ceriticate Authority (tls.crt)
+2. Private key associated with the TLS certificate(tls.key)
+3. Certificate Authority's (CA) root certificate(ca.crt)
+
+Create a secret in the same namespace as APK is deployed with the following key-value pairs:
+
+- tls.crt - Base64 encoded value of tls.crt file
+- tls.key - Base64 encoded value of tls.key file
+- ca.crt - Base64 encoded value of ca.crt file
+
+Update the `wso2.apk.dp.gatewayRuntime.deployment.router.configs` of values.yaml with the following values
+
+```
+configs:
+  tls:
+    secretName: "<Name of the created secret>"
+    certKeyFilename: "tls.key"
+    certFilename: "tls.crt"
+    certCAFilename: "ca.crt"
+```
+
+## Remove default IDP
+
+APK comes with a default IDP which is not production-ready. Disable the default IDP and use a production-ready IDP solution. Please follow these guidelines to  [setup the production ready IDP](https://apk.docs.wso2.com/en/latest/setup/identity-platform/idp/idp-overview/)
+
+Disable the default idp by changing the following value to `false` in values.yaml
+`idp.enabled` = `false`
+
+
+## Use a managed redis service
+
+APK uses a built-in standalone Redis service which is not suitable for production usage. Please use a managed, production-ready Redis. You can update the following values to configure the Redis configuration in APK:
+
+- wso2.apk.dp.redis.type
+- wso2.apk.dp.redis.url
+- wso2.apk.dp.redis.tls
+- wso2.apk.dp.redis.auth.certificatesSecret
+- wso2.apk.dp.redis.auth.secretKey
+- wso2.apk.dp.redis.poolSize
+
+
+## Protect gateway admin port
+
+APK uses EnvoyProxy in the router implementation. EnvoyProxy offers an administrator interface that can be used to query and modify different aspects of the server. In the production environment, we should disable or restrict access to this port. By default, APK exposes this interface through port `9000`. To disable external access to the port, you can set the following Helm value to `false`:
+`wso2.apk.dp.gatewayRuntime.deployment.router.adminInterfaceEnabled`
```

Comment:
When integrating with Prometheus we had to open this port for Prometheus. So we can give users two options. First case is to disable it. Second option is to let the port open and use network policies to restrict the access to the port.

Reply:
Instead of exposing the entire port, can't we ask the user to create an API for the specific envoy admin endpoint and use that for analytics purposes?

Reply:
If we disable this port, I dont think creating an API will work. You will get connection refused.

Reply:
Using this helm var we only disable the Kuberenetes service level port we are not disabling in the envoy level. [PR link](https://github.com/wso2/apk/pull/1759/files#diff-416216ebc54dd9e24f0189c36c106d3a67df0a9e7b323e4dbbd908ae422d2843R36)

User still have the ability to create a service that expose this port and they can create an API and expose this. 

Or should we completely disable this? (When I looked at their documentation i could not find a configuration that disable this admin feature.) 

Or should we just template [this](https://github.com/wso2/apk/blob/5242690d4e1ec5075a68b7feb46238ee9df55f04/helm-charts/templates/data-plane/gateway-components/gateway-runtime/gateway-runtime-deployment.yaml#L198) instead of service?

Reply:
I was under the impression that we disable this port in Choreo Connect as well. It looks like we can't disable it unless we change the bootstrap envoy.yaml. For this release we can do like this. We can remove the port completely from the K8s service. I dont see a real use case of having it there. For debugging purposes we have to port-forward to the port. For that we dont need the port to be available in the K8s service. So this config or any documentation step is not required. 

Right now we allow anyone to access this port when doing a port forward. We should restrict that to 127.0.0.1 as mentioned in the docs - https://www.envoyproxy.io/docs/envoy/latest/operations/admin

@tharindu1st @AmaliMatharaarachchi WDYT?

Reply:
Removed